packages/roosterjs-editor-api/lib/format/createLink.ts

```diff
@@ -66,42 +72,54 @@ export default function createLink(
+                        // Change text content if it is specified
+                        updateAnchorDisplayText(anchor, displayText);
+                    } else {
+                        anchor = editor.getDocument().createElement('A') as HTMLAnchorElement;
+                        anchor.textContent = displayText || originalUrl;
+                        anchor.href = normalizedUrl;
+                        editor.insertNode(anchor);
+                    }
                 } else {
-                    anchor = editor.getDocument().createElement('A') as HTMLAnchorElement;
-                    anchor.textContent = displayText || originalUrl;
-                    anchor.href = normalizedUrl;
-                    editor.insertNode(anchor);
-                }
-            } else {
-                // the selection is not collapsed, use browser execCommand
-                editor.getDocument().execCommand(DocumentCommand.CreateLink, false, normalizedUrl);
-                const traverser = editor.getSelectionTraverser();
+                    // the selection is not collapsed, use browser execCommand
+                    editor
+                        .getDocument()
+                        .execCommand(DocumentCommand.CreateLink, false, normalizedUrl);
+                    const traverser = editor.getSelectionTraverser();
 
-                let currentInline = traverser.getNextInlineElement();
+                    let currentInline = traverser.getNextInlineElement();
 
-                // list for removing unwanted lines
-                let deletionInlineList: Node[] = [];
+                    // list for removing unwanted lines
+                    let deletionInlineList: Node[] = [];
 
-                while (currentInline) {
-                    deletionInlineList.push(currentInline.getContainerNode());
-                    currentInline = traverser.getNextInlineElement();
-                }
+                    while (currentInline) {
+                        deletionInlineList.push(currentInline.getContainerNode());
+                        currentInline = traverser.getNextInlineElement();
+                    }
 
-                deletionInlineList.forEach(node => editor.deleteNode(node));
+                    deletionInlineList.forEach(node => editor.deleteNode(node));
 
-                anchor = getAnchorNodeAtCursor(editor);
-                updateAnchorDisplayText(anchor, displayText);
+                    anchor = getAnchorNodeAtCursor(editor);
+                    updateAnchorDisplayText(anchor, displayText);
+                }
+            } else if (selection.type === SelectionRangeTypes.ImageSelection) {
+                selection.ranges[0].deleteContents();
+                anchor = editor.getDocument().createElement('A') as HTMLAnchorElement;
+                anchor.appendChild(selection.image);
+                anchor.href = normalizedUrl;
+                editor.insertNode(anchor);

```

Comment:
You just need to wrap the image with a new A tag
const a = wrap(selection.imag, 'a');
a.href = ....

Reply:
True! I'll refactor!